src/hexen/a_action.c

```diff
@@ -1177,11 +1177,9 @@ void A_SoAExplode(mobj_t * actor)
     }
     if (actor->args[0])
     {                           // Spawn an item
-#if 0 // Checks are not present in version 1.1
-        if (!nomonsters
+        if ((gameversion != exe_hexen_1_1r2) || !nomonsters
```

Comment:
Is this the right logic here? I was expecting something like `if ((gameversion == exe_hexen_1_1r2) && (other conditions))`

Reply:
Yeah, these are the expected changes.
- In versions earlier than "1.1r2", having nonzero `actor->args[0]` is all that's needed for spawning an mobj of type `TranslateThingType[actor->args[0]]`, regardless of `-nomonsters` or the presence of the flag `MF_COUNTKILL`.
- As of "1.1r2", if, in addition, `-nomonsters` was passed, then the mobj won't be spawned if `MF_COUNTKILL` is set.

For reference, here is the recreated code: https://bitbucket.org/gamesrc-ver-recreation/hexen/src/1847583de60fdfe66476c30746740f0a6aa11dca/A_ACTION.C#lines-1327

```
	if(actor->args[0])
	{ // Spawn an item
#if (APPVER_HEXENREV >= AV_HR_HEX11B)
		if(!nomonsters 
		|| !(mobjinfo[TranslateThingType[actor->args[0]]].flags&MF_COUNTKILL))
#endif
		{ // Only spawn monsters if not -nomonsters
			P_SpawnMobj(actor->x, actor->y, actor->z,
				TranslateThingType[actor->args[0]]);
		}
	}
```

Reply:
Forced-pushed the branch. Marking this conversion (about the changes to a_action.c) as resolved.